### ISSUE TITLE
[JIT] Normalize representation of traced and scripted functions/modules 

### ISSUE THREAD

**Issue by jamesr66a** [pytorch/pytorch#8867]
Currently we have a mix of actual types when we script/trace a function/module:

```
import torch

@torch.jit.script
def script_fn(x):
    return x

print('*****Script function*****')
print(type(script_fn))
print(isinstance(script_fn, torch._C.GraphExecutor))

@torch.jit.trace(torch.zeros(3, 3))
def trace_fn(x):
    return x

print('*****Traced function*****')
print(type(trace_fn))
print(isinstance(trace_fn, torch._C.GraphExecutor))

class ScriptModule(torch.jit.ScriptModule):
    @torch.jit.script_method
    def forward(self, x):
        return x
sm = ScriptModule()

print('*****Script module*****')
print(type(sm))
print(isinstance(sm, torch._C.GraphExecutor))

class TracedModule(torch.nn.Module):
    def forward(self, x):
        return x
tm = torch.jit.trace(torch.zeros(3, 3))(TracedModule())

print('*****Traced module*****')
print(type(tm))
print(isinstance(tm, torch._C.GraphExecutor))

```

```
*****Script function*****
<class 'torch._C.GraphExecutor'>
True
*****Traced function*****
<class 'torch._C.GraphExecutor'>
True
*****Script module*****
<class '__main__.ScriptModule'>
False
*****Traced module*****
<class 'torch.jit.TopLevelTracedModule'>
False
```

This causes issues, for example, since we can only inline a `PythonValue` into a ScriptModule if it's an instance of GraphExecutor(https://github.com/pytorch/pytorch/blob/2b926aafb09575dd83ebf4fe2a76cbe239596f0b/torch/csrc/jit/script/init.cpp#L82). This causes problems in the case of calling Script/Traced modules from script functions -- they show up as `PythonOp`s.

Let's make it so that all of {script,traced} {function,module} produce a Module, and re-work the inlining logic to merge modules in all cases. This will greatly simplify the logic and make things easier to understand and maintain
